src/main/java/vimification/logic/commands/viewCommands/SearchByKeywordCommand.java

```diff
@@ -0,0 +1,24 @@
+package vimification.logic.commands.viewCommands;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import vimification.logic.commands.CommandException;
+import vimification.logic.commands.CommandResult;
+import vimification.model.LogicTaskList;
+import vimification.model.task.Task;
+
+public class SearchByKeywordCommand extends SearchCommand {
+
+    public SearchByKeywordCommand(String keyword) {
+        super(task -> task.containsKeyword(keyword));
+    }
+
+    @Override
+    public CommandResult execute(LogicTaskList taskList) throws CommandException {
+        ObservableList<Task> viewTaskList =
+                FXCollections.observableList(taskList.filter(getPredicate()));
```

Comment:
One thing I realized is that: the `ObservableList` created in `LogicManager` will share the same internal `List` with the `TaskList`.

Therefore, one way to implement this command is to just update the filter attached on the `ObservableList`.

I think we need more testing for this one.